src/kml/src/main/java/org/geoserver/kml/sequence/SequenceList.java

```diff
@@ -19,18 +19,15 @@
 
     public class GeneratorIterator implements Iterator<T> {
 
-        private Sequence<T> generator;
+        private Iterator<T> generator;
 
-        private T item;
-
-        public GeneratorIterator(Sequence<T> generator) {
+        public GeneratorIterator(Iterator<T> generator) {
             this.generator = generator;
-            this.item = generator.next();
```

Comment:
@aaime - the actual bugfix is here; previously the invocation of next() in the constructor would pre-emptively modify the KML context, leading to off-by-one errors on anything that referenced the current layer via the context.

Reply:
Makes sense. As said above, if Sequence goes the way of the dodo, the other classes need to be renamed too.

Reply:
I refactored everything, including the sequence package, renaming to iterator as appropriate. Confirmed the test passes in this branch and fails in master as well. Thanks for the notes let me know if anything else needs to be addressed.